src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2392,13 +2390,13 @@ PlanRouterQuery(Query *originalQuery,
 
 	/* both Postgres tables and materialized tables are locally avaliable */
 	RTEListProperties *rteProperties = GetRTEListPropertiesForQuery(originalQuery);
-	if (shardId == INVALID_SHARD_ID && ContainsOnlyLocalTables(rteProperties))
+
+	if (isLocalTableModification)
```

Comment:
Tbh, I cannot follow what is going on here? isLocalTableModification is a pointer, and seems to be expected to be filled by this function. How do we use it here?

and we have not changed any other function that is calling it.

Reply:
Yes, it's a pointer and _only_ when a valid pointer is passed, we fill it with true/false and eventually it will end up here

**Task->isLocalTableModification =**

in the planning phase. Later this flag is used in execution phase **CitusBeginModifyScan()**

Without our fix, if this flag is TRUE, we will skip generating shard-task-list for the reference as target, and source as Postgres table. In MERGE, this flag should be TRUE only if target is Postgres table.

Reply:
> Yes, it's a pointer, and only when a valid pointer is passed, we fill it with true/false which will eventually end up here in the planning phase.

Ok, I can follow this now. 

> In MERGE, this flag should be TRUE only if target is Postgres table.

As I said, let's try to understand why this is not a problem for UPDATE .. FROM type of queries a bit more. 

I'm hesitant to have a special case for Merge command, as it should (probably) not be different than UPDATE/DELETE with subqueries regarding this.